### ISSUE TITLE
Trying to figure out way to search multiple models - like 'threads' and 'replies'

### ISSUE THREAD

**Issue by TheBigK** [babenkoivan/scout-elasticsearch-driver#105]
This has to be straightforward, but I'm struggling to find out what's the right way to perform search on multiple models - like 'threads' and 'replies'. 

I'm trying to implement search across forum where the search is performed on following fields -

- thread -> title
- thread -> body
- reply -> body

The next challenge would be to rank these in order. For example, a keyword in thread title would be more important than in thread's body which would be more important than reply's body. Perhaps, I could prioritise the results by the date of creation in descending order.

Can someone point me in the right direction? 

Thank you in advance for your time. I really appreciate it.

**Comment by babenkoivan** [comment]
Hi @TheBigK, sorry for the late answer.
Have you managed to make up a solution for your problem?
I think that, you have to store all searchable information in the thread (title, body and replies).
If a new reply is added to a thread, you should update thread index in ES, by using `searchable` method.

**Comment by TheBigK** [comment]
Hi @babenkoivan - not really. I decided to settle for searching only the thread body and title. In future, I'm planning to turn my threads and replies into a single 'Posts' model so that I can perform search.

I'm wondering, if multi-index search is supported. If not, do you have any plans of adding it? It'd be RAD!

**Comment by babenkoivan** [comment]
Unfortunately, multi-index is not supported at the moment, but I'll think about possibility of adding it to a new version of the package.

**Comment by poncianodiego** [comment]
A possible solution could be using the toSearchableArray and only add the fields you require to search there?

**Comment by raymondtri** [comment]
@babenkoivan It seems that a multi-index is possible via elasticsearch https://www.elastic.co/guide/en/elasticsearch/reference/current/multi-index.html even to the point of wildcards.

Would creating a new "SearchableAll" model with the index name defined as '*' (and a scout prefix) be viable?

**Comment by raymondtri** [comment]
Alright so I found a solution to this, and since it appears to be surfacing for many people, I figured that I would post it here.

Essentially, you cannot use this specific package for multi-index searching. However, you can use the Elasticsearch/elasticsearch package which is a dependency of this package, and then manually write queries using their php sdk.

1. Specify the clientbuilder:
`use Elasticsearch\ClientBuilder;`
2. Instantiate the clientbuilder:
`$client = ClientBuilder::create()->setHosts(config('scout_elastic.client.hosts'))->build();`
3. Build your query using commas to separate index names
`
$params = [
    'index' => 'my_index_1,my_index_2,my_index_3',
    'body' => [
        'query' => [
            'bool' => [
                'must' => [
                    'match' => [
                        '_all' => 'Brazil'
                    ]
                ]
            ]
        ]
    ]
];
`
4. Get the response from elasticsearch
`
$response = $client->search($params);
`

**Comment by rognales** [comment]
>     'index' => 'my_index_1,my_index_2,my_index_3',

Do we lose Scout functionality this way?

**Comment by raymondtri** [comment]
Yes, you lose some functionality, but there is really no other way to do a multi-index search in this package that I have found.

**Comment by rognales** [comment]
I did try sharing the same index for multiple models (with standardize searchableArray) as a workaround but failed. As README stated, you need separator index for each model.
